src/cloudai/_core/installables/git_repo.py

```
@@ -0,0 +1,239 @@
+
+        submodules_res, submodules_msg = self.ensure_submodules_state(repo_path)
+        if not submodules_res:
+            logging.error(f"Submodule setup failed with `{submodules_msg}`, removing cloned repository at {repo_path}")
+            if repo_path.exists():
+                shutil.rmtree(repo_path)
+            return InstallStatusResult(False, submodules_msg)
+
+        return InstallStatusResult(True)
+
+    def _clone_repository(self, installer: "BaseInstaller", path: Path) -> InstallStatusResult:
+        logging.debug(f"Cloning repository {self.url} into {path}")
```

Ruff (0.15.12)

[warning] Logging statement uses f-string

(G004)